conans/model/pkg_type.py

```diff
@@ -58,6 +58,11 @@ def deduce_from_options():
                 if conanfile_type is PackageType.UNKNOWN:
                     raise ConanException(f"{conanfile}: Package type is 'library',"
                                          " but no 'shared' option declared")
+            elif any(option in conanfile.options for option in ["shared", "header_only"]):
```

Comment:
This happens for every package_type other than library, which I _think_ is valid, as `unknown` would still have precedence